packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -106,12 +106,77 @@ export type ClientResetConfig =
   | ClientResetRecoverUnsyncedChangesConfiguration
   | ClientResetRecoverOrDiscardUnsyncedChangesConfiguration;
 
+export type SSLConfiguration = {
+  /**
+   * Whether the SSL certificates must be validated. This should generally
+   * be `true` in production.
+   *
+   * Default is `true`.
+   */
+  validate?: boolean;
+  /**
+   * The path where to find trusted SSL certificates used to validate the
+   * server certificate. If `undefined`, validation will be delegated to
+   * the provided `validateCertificates` callback.
+   */
+  certificatePath?: string;
+  /**
+   * A callback function used to validate the server's SSL certificate. It
+   * is invoked for every certificate in the certificate chain starting from
+   * the root downward. An SSL connection will be established if all certificates
+   * are accepted. The certificate will be accepted if the callback returns `true`,
+   * or rejected if returning `false`. This callback is only invoked if
+   * `certificatePath` is `undefined`.
+   */
+  validateCertificates?: SSLVerifyCallback;
+};
+
+export type SSLVerifyCallback = (sslVerifyObject: SSLVerifyObject) => boolean;
+
+type SSLVerifyCallbackWithListArguments = (
+  serverAddress: string,
+  serverPort: number,
+  pemCertificate: string,
+  preverifyOk: number, // Acts as `SSLVerifyObject.acceptedByOpenSSL`
+  depth: number,
+) => boolean;
+
+export type SSLVerifyObject = {
```

Comment:
Not for this PR, but I think `SSLVerificationRequest` would probably be a better name for this.

Reply:
Or perhaps `Response` instead of `Request`? Since this is the object coming from the server.

We could definitely rethink this for v13 to not break the [earlier type](https://github.com/realm/realm-js/blob/a956a74a0845df99b6a4322627bc1c88a11ba749/types/index.d.ts#L133) for v12.

Reply:
> rethink this for v13 to not break the [earlier type](https://github.com/realm/realm-js/blob/a956a74a0845df99b6a4322627bc1c88a11ba749/types/index.d.ts#L133) for v12.

Yep :shipit: